packages/cli/src/commands/build/action.ts

```diff
@@ -0,0 +1,116 @@
+import { ConsolaInstance, createConsola, LogLevels } from 'consola';
+import { execa as _execa, Options as ExecaOptions } from 'execa';
+import { existsSync } from 'node:fs';
+import { join } from 'node:path';
+
+import { getProjectConfig } from '../../lib/project-config';
+
+import { opennextBuild } from './opennext-build';
+
+import { type Options as BuildCommandOptions } from './index';
+
+type Logger = Pick<ConsolaInstance, 'debug' | 'start' | 'success' | 'fail' | 'info' | 'error'>;
+type Config = () => {
+  get: (key: 'framework') => 'catalyst' | 'nextjs';
+  path: string;
+};
+export type Exec = (
+  command: string,
+  args?: string[],
+  options?: ExecaOptions,
+) => PromiseLike<unknown>;
+
+export function resolveFramework(options: BuildCommandOptions, getConfig: Config, logger: Logger) {
+  const config = getConfig();
+
+  if (options.verbose) {
+    logger.debug(`Reading config from ${config.path}`);
+
+    if (options.framework) {
+      logger.debug(`Using framework from --framework option: ${options.framework}`);
+    } else {
+      logger.debug(`Using framework from project configuration: ${config.get('framework')}`);
+    }
+  }
+
+  return options.framework ?? config.get('framework');
+}
+
+export function createLogger(options: BuildCommandOptions) {
+  return createConsola({ level: options.verbose ? LogLevels.verbose : LogLevels.info });
+}
+
+export function createExec(options: BuildCommandOptions, exec: Exec = _execa) {
```

Comment:
Not sure if these abstraction are worth it, can you provide your reasoning? Maybe they need to be lib utilities?